From efa7791bcd3a7f5c800719404e1b9f0455165a2b Mon Sep 17 00:00:00 2001
From: Joe Slater <jslater@windriver.com>
Date: Thu, 31 Jan 2019 11:13:14 +0800
Subject: [PATCH] refpolicy: insmod_t

modprobe might try to read /dev/null, and that will fail
in mls systems because insmod_t is s0-s15 while /dev/null
is s0.  So, we defeat that constraint for insmod_t.

Upstream-Status: Pending

Signed-off-by: Joe Slater <jslater@windriver.com>
Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/modutils.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/modutils.te b/policy/modules/system/modutils.te
index 8e61639..9aa734c 100644
--- a/policy/modules/system/modutils.te
+++ b/policy/modules/system/modutils.te
@@ -11,6 +11,7 @@ type kmod_t alias { insmod_t depmod_t update_modules_t };
 type kmod_exec_t alias { insmod_exec_t depmod_exec_t update_modules_exec_t };
 application_domain(kmod_t, kmod_exec_t)
 kernel_domtrans_to(kmod_t, kmod_exec_t)
+mls_file_read_all_levels(kmod_t)
 mls_file_write_all_levels(kmod_t)
 roleattribute system_r kmod_roles;
 role kmod_roles types kmod_t;
@@ -120,6 +121,9 @@ userdom_use_user_terminals(kmod_t)
 
 userdom_dontaudit_search_user_home_dirs(kmod_t)
 
+# read /dev/null !
+allow kmod_t device_t:chr_file read;
+
 ifdef(`init_systemd',`
 	# for /run/tmpfiles.d/kmod.conf
 	allow kmod_t kmod_tmpfiles_conf_t:file manage_file_perms;
-- 
2.7.4

